backend/services/stability_ai_service.py

```
@@ -25,6 +27,47 @@
+        
+        # Resize image
+        resized_image = image.resize((new_width, new_height), Image.Resampling.LANCZOS)
+        
+        # Convert back to bytes
+        output_bytes = io.BytesIO()
+        resized_image.save(output_bytes, format='PNG')
+        return output_bytes.getvalue()
+        
+    except Exception as e:
+        logger.error(f"Failed to resize image: {e}")
+        raise HTTPException(status_code=400, detail=f"Failed to process image: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)